core/src/main/clojure/xtdb/types.clj

```diff
@@ -1210,6 +1210,24 @@
                                (.get bb ba)
                                ba))}
 
+   :keyword {:typname "keyword"
+             :col-type :keyword
+             :oid 11111
```

Comment:
Arbitrarily chosen above 10000. It could be the case that this conflicts with existing types in other systems.